rebuild_slashing_protection.py

```diff
@@ -269,13 +274,11 @@ def process_chain(head, justified_epoch, processed_blocks, best_attestations_in_
 justified_epoch = int(finality_checkpoints["current_justified"]["epoch"])
 processed_blocks = []
 best_attestations_in_chains = []
+proposal_slots = []
 
 sorted(heads, key=lambda head: int(head["slot"]), reverse=True)
 for head in heads:
-    process_chain(head, justified_epoch, processed_blocks, best_attestations_in_chains, validator_index)
-    best_attestation = max(best_attestations_in_chains, key=lambda att: att.data.target.epoch)
-    if compute_epoch_at_slot(best_attestation.data.slot) > compute_epoch_at_slot(int(head["slot"])):
-        break
```

Comment:
removed this optimization as we ideally want to check every chain for a proposal

Reply:
Looks good. As a future desirable, we should stop the attestation search when a suitable attestation is found, so that we save on the expensive `cache_get_validator_committee_index()` calls. 

It computes `get_beacon_committee()` and checks the entire `committee` list for the `validator_index`, and this happens for every `(attestation_slot, attestation_index)` in this chain that we haven't computed it for already.